packages/fxa-auth-server/lib/payments/stripe.js

```diff
@@ -856,6 +867,137 @@ class StripeHelper {
       this.webhookSecret
     );
   }
+
+  /**
+   * Extract invoice details for billing emails
+   *
+   * @param {Invoice} invoice
+   */
+  async extractInvoiceDetailsForEmail(invoice) {
+    let uid = '',
+      email = '',
+      invoiceNumber = '',
+      invoiceDate,
+      invoiceTotal = 0,
+      cardType = '',
+      lastFour = '',
+      nextInvoiceDate,
+      productId = '',
+      productName = '',
+      planId = '',
+      planName = '',
+      planEmailIconURL = '',
+      planDownloadURL = '';
+
+    try {
+      if (invoice && typeof invoice === 'object') {
+        // Dig up & expand objects in the invoice that usually come as just IDs
+        const { plan } = invoice.lines.data[0];
+        const [abbrevProduct, customer, charge] = await Promise.all([
+          this.expandAbbrevProductForPlan(plan),
+          this.expandResource(invoice.customer, 'customers'),
+          this.expandResource(invoice.charge, 'charges'),
+        ]);
+
+        if (customer.deleted === true) {
+          throw error.unknownCustomer(invoice.customer);
+        }
+
+        const metadata = {
+          ...abbrevProduct.product_metadata,
+          ...plan.metadata,
+        };
+        uid = customer.metadata.userid;
+        ({ product_id: productId, product_name: productName } = abbrevProduct);
```

Comment:
Why did these get wrapped in a set of parens? The destructuring doesn't work otherwise?

Reply:
Yes, destructuring into local vars requires parens like this.

Reply:
`const { product_id: productId, product_name: productName } = abbrevProduct;`?

Reply:
No, these are being assigned to variables declared earlier with `let` https://github.com/mozilla/fxa/pull/4704/files#diff-26644f246b5f22f6205561269d539ce7R885